### PR TITLE
INTYGFV-15182: Changed so that we always use patient information from…

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
… PU when creating a new draft.

### CODE REVIEW

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice! A few comments to look at before its ready for merge.

File: web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java [hunk 1]
```diff
@@ -60,13 +58,34 @@ public CreateNewDraftRequest buildCreateNewDraftRequest(Intyg intyg, String inty
...
+    private Patient getPatient(Intyg intyg) {
+        final var pnr = intyg.getPatient().getPersonId().getExtension();
+        final var personnummer = Personnummer.createPersonnummer(pnr).orElseThrow();
+        return getPatientFromPU(personnummer);
+    }
+
+    private Patient getPatientFromPU(Personnummer personnummer) {
+        final var personFromPUService = patientDetailsResolver.getPersonFromPUService(personnummer);
+        Patient patient = new Patient();
+        patient.setPersonId(personFromPUService.getPerson().getPersonnummer());
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java:76]: Good suggestion, done.

File: web/src/test/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderTest.java [hunk 2]
```diff
@@ -80,11 +82,18 @@ public class CreateNewDraftRequestBuilderTest extends BaseCreateDraftCertificate
     @InjectMocks
     private CreateNewDraftRequestBuilderImpl builder;
 
+    private Person person;
+
     @Before
     public void setup() {
         user = buildWebCertUser();
         user.changeValdVardenhet(UNIT_HSAID);
+        person = new Person(Personnummer.createPersonnummer(PERSONNUMMER).get(), false, false, FORNAMN, MELLANNAMN, EFTERNAMN,
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply web/src/test/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderTest.java:98]: Added a test that verifies we use the information from PU.

File: web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java [hunk 3]
```diff
@@ -60,13 +58,34 @@ public CreateNewDraftRequest buildCreateNewDraftRequest(Intyg intyg, String inty
...
 
+    private Patient getPatient(Intyg intyg) {
+        final var pnr = intyg.getPatient().getPersonId().getExtension();
+        final var personnummer = Personnummer.createPersonnummer(pnr).orElseThrow();
+        return getPatientFromPU(personnummer);
+    }
+
+    private Patient getPatientFromPU(Personnummer personnummer) {
+        final var personFromPUService = patientDetailsResolver.getPersonFromPUService(personnummer).getPerson();
+        Patient patient = new Patient();
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java:75]: For consistency you could use final var ;-)

File: web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java [hunk 4]
```diff
@@ -60,13 +58,34 @@ public CreateNewDraftRequest buildCreateNewDraftRequest(Intyg intyg, String inty
         String intygsType = moduleRegistry.getModuleIdFromExternalId(intyg.getTypAvIntyg().getCode());
         Optional<Forifyllnad> forifyllnad = getOptionalForifyllnadIfApplicable(intygsType, intyg.getForifyllnad(), user);
 
-        final Patient patient = TransportConverterUtil.getPatient(intyg.getPatient(), PatientInfo.EXTENDED);
-        patient.setTestIndicator(patientDetailsResolver.isTestIndicator(patient.getPersonId()));
+        final Patient patient = getPatient(intyg);
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java:61]: For consistency you could use final var ;-)

**Review by mww-sopra-steria — APPROVED** [review]
Approved!

Noticed an inconsistency, feel free to do as you wish before merging :-)

File: web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java (hunk 3 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java]: Fixed!

File: web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java (hunk 4 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java]: Fixed

**Pull request merged**